waspc/data/Generator/templates/sdk/wasp/server/auth/hooks.ts

```diff
@@ -0,0 +1,29 @@
+import { type ProviderId } from '../../auth/utils.js'
+import { type PrismaClient } from '@prisma/client'
+
+export type AuthHookFn =
+  | OnUserSignUpHookFn
+  | OnUserLoginHookFn
+  | OnUserLogoutHookFn
+
+type OnUserLoginHookFn = (context: {
+  providerId: ProviderId
+  prisma: PrismaClient
+}) => Promise<void> | void
+type OnUserSignUpHookFn = (context: {
+  providerId: ProviderId
```

Comment:
Wondering, why not give them the whole User also? I guess they can fetch it on their own so there is no need?

Also, you are passing prisma. We normally either pass `entities`, or we let them import `prisma`. This is a third mechanism now -> why do this?